tests/extension_tests/deferred_bindings_tests/test_deferred_bindings.py

```diff
@@ -0,0 +1,122 @@
+            await host.init_worker()
+            r = await host.get_functions_metadata()
+            self.assertIsInstance(r.response, protos.FunctionMetadataResponse)
+            self.assertEqual(r.response.result.status,
+                             protos.StatusResult.Success)
+
+
+@unittest.skipIf(sys.version_info.minor <= 8, "The base extension"
+                                              "is only supported for 3.9+.")
+class TestDeferredBindingsEnabledDual(testutils.AsyncTestCase):
+
+    async def test_deferred_bindings_dual_metadata(self):
+        async with testutils.start_mockhost(
+                script_root=DEFERRED_BINDINGS_ENABLED_DUAL_DIR) as host:
+            await host.init_worker()
+            r = await host.get_functions_metadata()
+            self.assertIsInstance(r.response, protos.FunctionMetadataResponse)
+            self.assertEqual(r.response.result.status,
+                             protos.StatusResult.Success)
+
+
+@unittest.skipIf(sys.version_info.minor <= 8, "The base extension"
+                                              "is only supported for 3.9+.")
+class TestDeferredBindingsDisabled(testutils.AsyncTestCase):
+
+    async def test_non_deferred_bindings_metadata(self):
+        async with testutils.start_mockhost(
+                script_root=DEFERRED_BINDINGS_DISABLED_DIR) as host:
+            await host.init_worker()
+            r = await host.get_functions_metadata()
+            self.assertIsInstance(r.response, protos.FunctionMetadataResponse)
+            self.assertEqual(r.response.result.status,
+                             protos.StatusResult.Success)
+
+
+@unittest.skipIf(sys.version_info.minor <= 8, "The base extension"
+                                              "is only supported for 3.9+.")
+class TestDeferredBindingsHelpers(testutils.AsyncTestCase):
+
+    def test_deferred_bindings_decode(self):
+        binding = BlobClientConverter
+        pb = protos.ParameterBinding(name='test',
+                                     data=protos.TypedData(
+                                         string='test'))
+        sample_mbd = MockMBD(version="1.0",
+                             source="AzureStorageBlobs",
+                             content_type="application/json",
+                             content="{\"Connection\":\"AzureWebJobsStorage\","
+                                     "\"ContainerName\":"
+                                     "\"python-worker-tests\","
+                                     "\"BlobName\":"
+                                     "\"test-blobclient-trigger.txt\"}")
+        datum = datumdef.Datum(value=sample_mbd, type='model_binding_data')
+
+        obj = meta.deferred_bindings_decode(binding=binding, pb=pb,
+                                            pytype=BlobClient, datum=datum, metadata={})
+
+        self.assertIsNotNone(obj)
+
+    async def test_check_deferred_bindings_enabled(self):
```

Comment:
What is the tuple here? Add a pydoc to explain this test. This definitely needs to be solved.

Reply:
It's a tuple representing if deferred bindings is supported at the app level and for that specific binding. The method has pydocs, I'll add to this test as well